### PR TITLE
[logging] add bbr region

### PR DESCRIPTION
**Pull request by librasungirl**
This PR depends on #4795 , please only review the last commit which includes the update.

### CODE REVIEW

File: src/core/backbone_router/leader.cpp [hunk 1]
```diff
@@ -87,47 +83,56 @@ otError Leader::GetServiceId(uint8_t &aServiceId) const
...
     OT_UNUSED_VARIABLE(aConfig);
 
-    otLogInfoNetData("BBR state %s", StateToString(aState));
+    otLogInfoBbr("PBBR state: %s", StateToString(aState));
 
     if (aState != kStateRemoved && aState != kStateNone)
     {
-        otLogInfoNetData("Rloc16: 0x%4X, seqno: %d, delay: %d, timeout %d", aConfig.mServer16, aConfig.mSequenceNumber,
-                         aConfig.mReregistrationDelay, aConfig.mMlrTimeout);
+        otLogInfoBbr("Rloc16: 0x%4X, seqno: %d, delay: %d, timeout %d", aConfig.mServer16, aConfig.mSequenceNumber,
```
Comment (simonlingoogle) [inline src/core/backbone_router/leader.cpp:95]: merge logs to one line ?

Comment (librasungirl) [reply src/core/backbone_router/leader.cpp:95]: This PR serves only for logging update, I'd prefer not to introduce unrelated changes into this one.

Besides, regarding the comment, there is no valid dataset when the state is `kStateRemoved` or `kStateNone`,  one line for all scenarios would make it not easy to read the code in my view.

Any suggestion to achieve one line log? @simonlingoogle

**Comment by librasungirl** [comment]
rebased as the dependent #4795 gets merged.

File: src/core/backbone_router/leader.cpp (hunk 1 above)
Comment (simonlingoogle) [reply src/core/backbone_router/leader.cpp:95]: We can simply use `if else` with 2 different lines of `otLogInfoBbr`, but only one will be executed at runtime.

Comment (librasungirl) [reply src/core/backbone_router/leader.cpp:95]: It is an option. No strong opinion. Anyway feel free to submit a PR for it if you prefer this way. Thanks.

Will mark this as resolved as it is not related to this PR.

File: src/core/common/logging.hpp [hunk 2]
```diff
@@ -89,6 +89,9 @@ extern "C" {
 #define _OT_REGION_CLI_PREFIX "-CLI-----: "
 #define _OT_REGION_CORE_PREFIX "-CORE----: "
 #define _OT_REGION_UTIL_PREFIX "-UTIL----: "
+#if OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2
```
Comment (jwhui) [inline src/core/common/logging.hpp:92]: nit: how about we avoid feature-flagging these defines (here and below)?

Comment (librasungirl) [reply src/core/common/logging.hpp]: removed.

**Pull request merged**